### PR TITLE
Only show the "View all work" button if there are at least 7 items.

### PR DESCRIPTION
**Pull request by KaiPrince**
## Overview

6 or less | 7 or more
-------|-------
image | image

Only show the "View all work" button if there are at least 7 items.

### Reason for change

See ticket: [DE43688](https://rally1.rallydev.com/#/515253241784ud/workviews?detail=%2Fdefect%2F605810430964&fdp=true?fdp=true): W2D widget on course page always shows View All Work link

### Changes summarized

- Add conditional to "View all work" button to only display if page total is greater than page size.
- Add a test that will fail if this behaviour stops.

### Suggested future tasks

None.

## Checklist

- [x] No new warnings are introduced
- [x] Comments and documentation are up to date
- [x] TODOs resolved or added to issue tracker
- [x] No commented-out code
- [x] This isn't a 'quick-and-dirty' job

### CODE REVIEW

File: test/data/workToDo.js [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+						},
+					],
+				},
+			],
+		},
+	],
+	links: [
+		{
+			rel: ['https://api.brightspace.com/rels/my-organization-awards'],
+			href: 'https://2b9c692b-7fae-4557-bc79-66af9b71e0fc.awards.api.proddev.d2l/organizations/6606/users/10183?allOrgUnits=1',
```
Comment (lloydwaddell) [inline test/data/workToDo.js:61]: nit: could cleanup links that aren't being used to help readability

File: test/data/workToDo.js [hunk 2]
```diff
@@ -0,0 +1,665 @@
...
+		},
+		{
+			rel: ['https://api.brightspace.com/rels/feedback'],
+			href: 'https://2b9c692b-7fae-4557-bc79-66af9b71e0fc.feedback.api.proddev.d2l/',
+		},
+	],
+};
+
+// https://2b9c692b-7fae-4557-bc79-66af9b71e0fc.organizations.api.proddev.d2l/6606 => /w2d-orgs
+// https://2b9c692b-7fae-4557-bc79-66af9b71e0fc.organizations.api.proddev.d2l/6609 => /w2d-orgs-2
```
Comment (lloydwaddell) [inline test/data/workToDo.js:334]: nit: not sure comments are adding anything here, maybe leave it at "mock response from organizations api /w2d-orgs" ?

File: features/workToDo/test/d2l-w2d-work-to-do.test.js [hunk 3]
```diff
@@ -16,4 +51,65 @@ describe('d2l-w2d-work-to-do', () => {
...
+			mockLink.mock('path:/w2d-activity-6', workToDoActivity);
+			mockLink.mock('path:/w2d-activity-7', workToDoActivity);
+			mockLink.mock('path:/w2d-activity-8', workToDoActivity);
+			mockLink.mock('path:/w2d-activity-9', workToDoActivity);
+			mockLink.mock('path:/w2d-activity-10', workToDoActivity);
+			mockLink.mock('path:/w2d-folder4', workToDoFolder);
+		});
+
+		afterEach(() => {
+			mockLink.reset();
```
Comment (lloydwaddell) [inline features/workToDo/test/d2l-w2d-work-to-do.test.js:76]: nit: this is being called in the beforeEach, so only really needs to be in an after to capture the last test.

File: features/workToDo/d2l-w2d-collections.js [hunk 4]
```diff
@@ -291,7 +291,7 @@ class W2dCollections extends LocalizeDynamicMixin(HypermediaStateMixin(LitElemen
 				${overdue}
 				${categories && categories.length > 0 ? this._renderHeader2(this.localize('upcoming'), this._pagingTotalResultsUpcoming) : null}
 				${categories}
-				${this.dataFullPagePath && this._loaded && this.collapsed ? html`<d2l-link href="${this.dataFullPagePath}">${this.localize('fullViewLink')}</d2l-link>` : null}
+				${this.dataFullPagePath && this._loaded && this.collapsed && this._pagingTotalResultsOverdue > pageSize.collapsed ? html`<d2l-link href="${this.dataFullPagePath}">${this.localize('fullViewLink')}</d2l-link>` : null}
```
Comment (lloydwaddell) [inline features/workToDo/d2l-w2d-collections.js:294]: is this only checking the overdue items? what happens if there are a mix of overdue and upcoming?

Comment (KaiPrince) [reply features/workToDo/d2l-w2d-collections.js:294]: Oh shoot that I gotta fix

File: features/workToDo/test/d2l-w2d-work-to-do.test.js (hunk 3 above)
Comment (KaiPrince) [reply features/workToDo/test/d2l-w2d-work-to-do.test.js:76]: Fixed with https://github.com/BrightspaceHypermediaComponents/foundation-components/pull/259/commits/0d36e19ecda8a44b0f31faa6a5979407fe1d8682

File: test/data/workToDo.js (hunk 1 above)
Comment (KaiPrince) [reply test/data/workToDo.js]: Fixed with https://github.com/BrightspaceHypermediaComponents/foundation-components/pull/259/commits/a1be5fc925c83d92a9be9b58ca3859f62790e1be

File: test/data/workToDo.js (hunk 2 above)
Comment (KaiPrince) [reply test/data/workToDo.js]: Fixed with https://github.com/BrightspaceHypermediaComponents/foundation-components/pull/259/commits/a1be5fc925c83d92a9be9b58ca3859f62790e1be

File: features/workToDo/test/d2l-w2d-work-to-do.test.js [hunk 5]
```diff
@@ -16,4 +51,65 @@ describe('d2l-w2d-work-to-do', () => {
...
+			mockLink.mock('path:/w2d-activity-9', workToDoActivity);
+			mockLink.mock('path:/w2d-activity-10', workToDoActivity);
+			mockLink.mock('path:/w2d-folder4', workToDoFolder);
+		});
+
+		after(() => {
+			mockLink.reset();
+		});
+
+		it('does not show if 6 or fewer items', async() => {
```
Comment (lloydwaddell) [inline features/workToDo/test/d2l-w2d-work-to-do.test.js:79]: nit: no test for mix of overdue / upcoming activities, but I'm also ok with not crowding the repo with tests. Just be mindful that not all scenarios are covered in unit tests

**Review by lloydwaddell — APPROVED** [review]

**Pull request merged**